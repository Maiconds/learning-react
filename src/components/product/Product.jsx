import React from 'react';
import PropTypes from 'prop-types';

const Product = ({ name, price, description, showPrice }) => (
  <div>
    <p className="product-name">{name}</p>
    { showPrice ? <p className="product-price">{price}</p> : null}
    <p className="product-description">{description}</p>
  </div>
);

Product.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  description: PropTypes.string,
  showPrice: PropTypes.bool.isRequired
};

export default Product; 
