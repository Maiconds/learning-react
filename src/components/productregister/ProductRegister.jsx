import  React, { Component }  from 'react';
import Product from '../product/Product';
import { getAllProducts } from '../../services/productServices';

class ProductRegister extends Component {
    constructor(props){
        super(props);

        this.state = {
            name: '',
            price: 0,
            description: '',
            products: []
        }
        
        this.change = this.change.bind(this);
        this.save = this.save.bind(this);

    }

    componentDidMount(){
        getAllProducts().then( 
            json => this.setState({
                products: json
            })
        ).catch(window.console.log("Erro ao carregar dados"))
    }

    change(e, field) {
        e.preventDefault();
        const value = e.target.value;
        const isPrice = field === 'price';
        if(!(isNaN(value) && isPrice)){
            this.setState({
                [field]: field==='price'? value==='' ? 0 : parseFloat(value) : value         
            })
        }      
    }

    save(){
        const {name, price, description} = this.state;

        if(name !== '' && price!==0){
            this.setState(state => ({
                name: '',
                price: 0,
                description: '',
                products: [...state.products, { name, price, description }]
              }));
        }
    }

    render(){
        const {name, price, description, products} = this.state;
        return(
            <div>
               Nome: <input type="text" name="name" value={name} onChange={e => this.change(e, 'name')} />
               Price: <input type="text" value={price} name="price" onChange={e => this.change(e, 'price')}/>
               Description: <input type="text" value={description} name="description" onChange={e => this.change(e, 'description')}/><br/><br/>
               <button onClick={this.save}>Salvar</button>
               <h1>Products List:</h1>
               {products.map(product => <Product name={product.name} price={product.price} description={product.description}/>)}
            </div>
        );
    }
}

export default ProductRegister;
