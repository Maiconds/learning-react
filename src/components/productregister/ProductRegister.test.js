import React from 'react';
import { shallow } from 'enzyme';
import Product from './ProductRegister';
import ProductRegister from './ProductRegister';

describe('ProductRegister component test', () => {
    
    describe('Snapshot test', () => {
        
        const component = shallow(<ProductRegister/>);

        it('Should mactch the snapshot', () => {
            expect(component).toMatchSnapshot();
          }); 
    })
});