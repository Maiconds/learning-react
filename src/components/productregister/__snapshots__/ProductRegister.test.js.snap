// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductRegister component test Snapshot test Should mactch the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProductRegister />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Nome: ",
        <input
          name="name"
          onChange={[Function]}
          type="text"
          value=""
        />,
        "Price: ",
        <input
          name="price"
          onChange={[Function]}
          type="text"
          value={0}
        />,
        "Description: ",
        <input
          name="description"
          onChange={[Function]}
          type="text"
          value=""
        />,
        "Filter: ",
        <input
          name="filter"
          onChange={[Function]}
          type="text"
          value=""
        />,
        "Show Prices: ",
        <input
          checked={true}
          name="showPrices"
          onChange={[Function]}
          type="checkbox"
        />,
        <br />,
        <br />,
        <button
          onClick={[Function]}
        >
          Salvar
        </button>,
        <h1>
          Products List:
        </h1>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      "Nome: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "name",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Price: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "price",
          "onChange": [Function],
          "type": "text",
          "value": 0,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Description: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "description",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Filter: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "name": "filter",
          "onChange": [Function],
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "Show Prices: ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "name": "showPrices",
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Salvar",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Salvar",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Products List:",
        },
        "ref": null,
        "rendered": "Products List:",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Nome: ",
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />,
          "Price: ",
          <input
            name="price"
            onChange={[Function]}
            type="text"
            value={0}
          />,
          "Description: ",
          <input
            name="description"
            onChange={[Function]}
            type="text"
            value=""
          />,
          "Filter: ",
          <input
            name="filter"
            onChange={[Function]}
            type="text"
            value=""
          />,
          "Show Prices: ",
          <input
            checked={true}
            name="showPrices"
            onChange={[Function]}
            type="checkbox"
          />,
          <br />,
          <br />,
          <button
            onClick={[Function]}
          >
            Salvar
          </button>,
          <h1>
            Products List:
          </h1>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        "Nome: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Price: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "price",
            "onChange": [Function],
            "type": "text",
            "value": 0,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Description: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "description",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Filter: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "filter",
            "onChange": [Function],
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "Show Prices: ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "name": "showPrices",
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Salvar",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Salvar",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Products List:",
          },
          "ref": null,
          "rendered": "Products List:",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
